### ISSUE TITLE
WAI-ARIA rule

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#113]
I designed a rule for WAI-ARIA is:

* * *

Warn if the `role` attribute and/or `aria-*` attributes don't set in accordance with specs that are WAI-ARIA and/or "ARIA in HTML".

Warn if:

-   Use the role that doesn't exist in the spec.
-   Use the abstract role.
-   Use the `aria-*` attribute that doesn't belong to a set role (or an implicit role).
-   [Optional] Use a bad value of the `aria-*` attribute.
-   [Optional] Use the not permitted role according to ARIA in HTML.
-   [Optional] Set the implicit role explicitly.

## Rule Details

👎 Examples of **incorrect** code for this rule

```html
<div role="landmark" aria-busy="busy">
	<ul>
		<li role="button">an item</li>
	</ul>
	<button aria-checked="true">Click me!</button>
</div>
```

👍 Examples of **correct** code for this rule

```html
<div role="banner" aria-busy="true">
	<ul>
		<li role="menuitemcheckbox">an item</li>
	</ul>
	<button aria-pressed="true">Click me!</button>
</div>
```

### Setting value

Type: `boolean`

#### `checkingValue`

Warn if use a bad value of the `aria-*` attribute. This is optional but you should not disable it if you need to follow the spec. But you can disable it temporarily if you need when it happened to a problem on the spec was updated (ex: If a value was added to an allowed list on the spec, etc).

-   Type: `boolean`
-   Optional
-   Default: `true`

##### `permittedAriaRoles`

Warn if use the not permitted role according to ARIA in HTML. This is optional because it is for the HTML as the host language and it is not the spec of WAI-ARIA.

-   Type: `boolean`
-   Optional
-   Default: `true`

##### `disallowSetImplicitRole`

Disallow set the implicit role explicitly. This is optional because it is for the HTML as the host language and it is not the spec of WAI-ARIA.

-   Type: `boolean`
-   Optional
-   Default: `true`

### Default notification severity

`error`

**Comment by YusukeHirao** [comment]
@masuP9 

The rules for ARIA in HTML become optional because ARIA in HTML and WAI-ARIA are different specifications.
Please comment if you have any opinions.

**Comment by masuP9** [comment]
@YusukeHirao Sorry for late comment. It looks good! But how about make to use a bad value of the aria-* attribute non-Optional. It's the WAI-ARIA 

That's because this rule is a defined specification by only WAI-ARIA and not a rule as to its relationship to the host language.  The other optional rules seem to be in relation to host languages.

**Comment by YusukeHirao** [comment]
@masuP9 
Thanks for checking and advice.

> That's because this rule is a defined specification by only WAI-ARIA and not a rule as to its relationship to the host language. The other optional rules seem to be in relation to host languages.

True, I might need to unify the kind of the optional.

But I think checking to bad value might need to be optional because a developer wants to disable it temporarily if the spec updated (ex: If a value was added to an allowed list).

So I add a reason for the above to the doc.

I continue improving PR, Thank you.

**Comment by YusukeHirao** [comment]
I added the below:

> This is optional but you should not disable it if you need to follow the spec. But you can disable it temporarily if you need when it happened to a problem on the spec was updated (ex: If a value was added to an allowed list on the spec, etc).

**Comment by YusukeHirao** [comment]
I also added the below:

> This is optional because it is for the HTML as the host language and it is not the spec of WAI-ARIA.
